include/crypto/mlkem.h

```diff
@@ -75,105 +73,155 @@ extern "C" {
+ */
+int ossl_mlkem768_encap(uint8_t *out_ciphertext,
+                        uint8_t *out_shared_secret,
+                        const ossl_mlkem768_public_key *public_key,
+                        ossl_mlkem_ctx *mlkem_ctx);
 
-# endif /* OPENSSL_NO_MLKEM */
+/*
+ * MLKEM_ENCAP_ENTROPY is the number of bytes of uniformly random entropy
+ * necessary to encapsulate a secret. The entropy will be leaked to the
+ * decapsulating party.
+ */
+#  define MLKEM_ENCAP_ENTROPY 32
+
+/*
+ * The same as ossl_mlkem768_encap except this also uses a given entropy for
+ * deterministic output.
+ */
+int ossl_mlkem768_encap_external_entropy(uint8_t *out_ciphertext,
+                                         uint8_t *out_shared_secret,
+                                         const ossl_mlkem768_public_key *public_key,
+                                         const uint8_t *entropy,
+                                         ossl_mlkem_ctx *mlkem_ctx);
+
+/*
+ * ossl_mlkem768_decap decrypts a shared secret from |ciphertext| using
+ * |private_key| and writes it to |out_shared_secret|. If |ciphertext_len| is
+ * incorrect it returns 0, otherwise it returns 1. If |ciphertext| is invalid
+ * (but of the correct length), |out_shared_secret| is filled with a key that
+ * will always be the same for the same |ciphertext| and |private_key|, but
+ * which appears to be random unless one has access to |private_key|. These
+ * alternatives occur in constant time. Any subsequent symmetric encryption
+ * using |out_shared_secret| must use an authenticated encryption scheme in
+ * order to discover the decapsulation failure. it is assumed out_shared_secret
+ * has been allocated MLKEM_SHARED_SECRET_BYTES bytes
+ */
+int ossl_mlkem768_decap(uint8_t *out_shared_secret,
+                        const uint8_t *ciphertext, size_t ciphertext_len,
+                        const ossl_mlkem768_private_key *private_key,
+                        ossl_mlkem_ctx *mlkem_ctx);
+
+/* Serialisation of keys. */
+
+/*
+ * ossl_mlkem768_marshal_public_key serializes |public_key| to |out| in the
+ * standard format for ML-KEM-768 public keys. It returns one on success or zero
+ * on allocation error.
+ */
+int ossl_mlkem768_marshal_public_key(uint8_t *out,
+                                     const struct ossl_mlkem768_public_key *public_key);
 
-# if defined(__cplusplus)
-}  /* extern C */
-# endif
+/*
+ * ossl_mlkem768_parse_public_key parses a public key, in the format generated
+ * by |ossl_mlkem768_marshal_public_key|, from |in| and writes the result to
+ * |out_public_key|. It returns one on success or zero on parse error.
+ */
+int ossl_mlkem768_parse_public_key(struct ossl_mlkem768_public_key *out_public_key,
+                                   uint8_t *in,
+                                   ossl_mlkem_ctx *mlkem_ctx);
+
+/*
+ * MLKEM768_PRIVATE_KEY_BYTES is the length of the data produced by
+ * |MLKEM768_marshal_private_key|.
+ */
+#  define ossl_mlkem768_PRIVATE_KEY_BYTES 2400
```

Comment:
Nit: Please use all UPPERCASE for macros.

Reply:
Done, and I've also renamed MLKEM_SEED_BYTES to OSSL_MLKEM_SEED_BYTES